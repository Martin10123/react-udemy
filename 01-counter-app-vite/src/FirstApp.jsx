import PropTypes from "prop-types";

const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid= "test-title">{title}</h1>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  name: "Martin",
  subTitle: "1234",
  // title: ""
}

export default FirstApp;
