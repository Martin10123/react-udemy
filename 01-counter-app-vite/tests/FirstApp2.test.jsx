import { render, screen } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe("Pruebas en el component FirstApp", () => {
  const title = "Holaaaaaaaa";
  const subTitle = "Hola, soy un subtitulo";

  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar el mensaje de la variable title", () => {
    render(<FirstApp title={title} />);

    expect(screen.getByText(title)).toBeTruthy();
  });

  test("Debe de mostrar el titulo en un h1", () => {
    render(<FirstApp title={title} />);

    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("Debe de mostrar el subtitulo enviado por props", () => {
    render(<FirstApp title={title} subTitle={subTitle} />);

    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});
