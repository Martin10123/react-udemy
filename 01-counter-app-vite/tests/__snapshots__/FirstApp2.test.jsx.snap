// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el component FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Holaaaaaaaa
  </h1>
  <p>
    1234
  </p>
  <p>
    1234
  </p>
  <p>
    Martin
  </p>
</div>
`;
